src/Shared/Memoization/Memoize.cs

```diff
@@ -75,6 +76,8 @@ internal static class Memoize
         "Major Code Smell",
         "S2436:Types and methods should not have too many generic parameters",
         Justification = "We're using many generic types for the same reason Func<>, Func<,>, Func<,,>, ... exist.")]
-    public static Func<TParameter1, TParameter2, TParameter3, TResult> Function<TParameter1, TParameter2, TParameter3, TResult>(Func<TParameter1, TParameter2, TParameter3, TResult> f)
+    public static Func<TParameter1, TParameter2, TParameter3, TResult>
```

Comment:
I wasn't too sure about the formatting on this once, since it's now ridiculously long and I was getting a warning

Reply:
This code (the 3 parameter function) doesn't appear to be used. @geeknoid - thoughts on just deleting it?

Reply:
If it's not used, nuke it.

Reply:
Thanks. I've removed it.